src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,27 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        Rectangle current = (Rectangle) obj;
+        return (this.width == null || this.width.equals(current.width))
+                && (this.length == null || this.length.equals(current.length))
+                && (this.color == null || this.color.equals(current.color));
```

Comment:
are you sure that this check is ok?

Reply:
we can't invoke equals() on null object, so we need to check for it